sdk/eventhub/Microsoft.Azure.EventHubs/tests/Processor/ProcessorTests.cs

```diff
@@ -0,0 +1,1190 @@
+
+                var epo = await GetOptionsAsync(connectionString);
+
+                // Run a generic scenario with TestEventProcessor instead
+                await RunGenericScenario(eventProcessorHost, epo);
+            }
+        }
+
+        [Fact]
+        [LiveTest]
+        [DisplayTestMethodName]
+        public async Task ReRegisterAfterLeaseExpiry()
+        {
+            await using (var scope = await EventHubScope.CreateAsync(2))
+            {
+                var connectionString = TestUtility.BuildEventHubsConnectionString(scope.EventHubName);
+                var hostName = Guid.NewGuid().ToString();
+
+                var processorOptions = new EventProcessorOptions
+                {
+                    ReceiveTimeout = TimeSpan.FromSeconds(10),
+                    InitialOffsetProvider = pId => EventPosition.FromEnd()
+                };
+
+                var eventProcessorHost = new EventProcessorHost(
+                    hostName,
+                    null, // Entity path will be picked from connection string.
+                    PartitionReceiver.DefaultConsumerGroupName,
+                    connectionString,
+                    TestUtility.StorageConnectionString,
+                    Guid.NewGuid().ToString());
+
+                var runResult = await RunGenericScenario(eventProcessorHost, processorOptions);
+                Assert.False(runResult.ReceivedEvents.Any(kvp => kvp.Value.Count != 1), "First host: One of the partitions didn't return exactly 1 event");
+
+                // Allow sometime so that leases can expire.
+                await Task.Delay(60);
+
+                runResult = await RunGenericScenario(eventProcessorHost);
+                Assert.False(runResult.ReceivedEvents.Any(kvp => kvp.Value.Count != 1), "Second host: One of the partitions didn't return exactly 1 event");
+            }
+        }
+
+        [Fact]
+        [LiveTest]
+        [DisplayTestMethodName]
+        public async Task LargeHostName()
+        {
+            await using (var scope = await EventHubScope.CreateAsync(2))
+            {
+                var longHostname = StringUtility.GetRandomString(100);
+                var connectionString = TestUtility.BuildEventHubsConnectionString(scope.EventHubName);
+                var eventProcessorHost = new EventProcessorHost(
+                    longHostname,
+                    null, // Entity path will be picked from connection string.
+                    PartitionReceiver.DefaultConsumerGroupName,
+                    connectionString,
+                    TestUtility.StorageConnectionString,
+                    Guid.NewGuid().ToString());
+
+                var epo = await GetOptionsAsync(connectionString);
+                await RunGenericScenario(eventProcessorHost, epo);
+            }
+        }
+
+        [Fact]
+        [LiveTest]
+        [DisplayTestMethodName]
+        public async Task DontCheckpointStartOfStream()
```

Comment:
>public async Task DontCheckpointStartOfStream() [](start = 8, length = 47)

This is the new test added in this PR, rest is move from base class to child class.